arch/src/x86_64/smbios.rs

```diff
@@ -157,7 +157,7 @@ fn write_string(
     for c in val.as_bytes().iter() {
         curptr = write_and_incr(mem, *c, curptr)?;
     }
-    curptr = write_and_incr(mem, 0 as u8, curptr)?;
+    curptr = write_and_incr(mem, 0u8, curptr)?;
```

Comment:
Is there a reason you didn't have this type of change the avoid cast PR?

Reply:
We need to keep the specific type here a the function takes a T and we need it to be 1 byte only.

Reply:
You're right though, this shouldn't be in this commit.